src/main/java/org/example/Animal.java

```diff
@@ -1,4 +1,54 @@
 package org.example;
 
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Data;
+import lombok.Getter;
+
+import java.util.Objects;
+
+@Getter
+@Data
+
 public class Animal {
-}
+
+    private final String name;
+    private final String breed;
+
+    private final int age;
+
+    @JsonCreator
+    public Animal(
+            @JsonProperty("name") String name,
+            @JsonProperty("breed") String breed,
+            @JsonProperty("age") int age) {
+        this.name = name;
+        this.breed = breed;
+        this.age = age;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Animal animal = (Animal) o;
+        return age == animal.age &&
+                Objects.equals(name, animal.name) &&
+                Objects.equals(breed, animal.breed);
+    }
+
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(name, breed, age);
+    }
+
+    @Override
+    public String toString() {
```

Comment:
As you add Lombok dependency, you can use its annotations to generate equals, hashcode and tostring (in case you do not use any specific implementation)